### PR TITLE
Add devise user

### PR DESCRIPTION
**Pull request by alanluqman**
## Description
In this PR, I have installed Devise for user model. all user can :

### Can register a new user.

sign up
## 

### User logs in with a combination of email and password.

login

user

##

### User can reset password.
by clicking username in navbar

edit password

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
```
Comment (mirouhml) [inline app/models/user.rb]: - [x] Please know that the application should ask for confirmation of the email, you can do that by adding `:confirmable` here, and adjusting another setting I'll mention below.

File: db/migrate/20221214090559_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      # Uncomment below if timestamps were not included in your original model.
+      # t.timestamps null: false
+    end
+
+    add_index :users, :email, unique: true
+    add_index :users, :reset_password_token, unique: true
+    # add_index :users, :confirmation_token,   unique: true
```
Comment (mirouhml) [inline db/migrate/20221214090559_add_devise_to_users.rb]: - [x] To enable confirmation email as I mentioned above you'll need to uncomment this line as well and then redo your migration.

**Review by mirouhml — CHANGES_REQUESTED** [review]
# REQUIRED CHANGES 🔴

Hi @alanluqman, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221214090559_add_devise_to_users.rb (hunk 2 above)
Comment (alanluqman) [reply db/migrate/20221214090559_add_devise_to_users.rb]: **Dear @mirouhml , I tried what you said , but it does not allow user to log in ! however it registered before. So I decided to remove that confirmable to let the app work**
confirm

File: app/models/user.rb (hunk 1 above)
Comment (DeliceLydia) [reply app/models/user.rb]: - [x] This should be addressed as suggested with the previous reviewer

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @alanluqman, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by danmainah — APPROVED** [review]
Hi @alanluqman , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Github flow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**